fix(makefile): remove package-lock.json as install stamp prerequisite

## Problem

`make tf-plan` (and any target that depends on `$(LAMBDAS_STAMP)`) fails on a fresh clone with:

```
make: *** No rule to make target 'app/package-lock.json', needed by '.make/install.stamp'.  Stop.
```

`app/package-lock.json` is listed in `PKG_JSONS` as a hard prerequisite for the install stamp:

```makefile
PKG_JSONS    := $(shell find $(APP_DIR)/packages -name 'package.json' 2>/dev/null) \
                $(APP_DIR)/package.json \
                $(APP_DIR)/package-lock.json   # ← problem
```

The file is not committed to the repo and not gitignored, so it doesn't exist until after `npm install` runs — which is exactly what the stamp is supposed to trigger. Make sees a missing prerequisite with no rule to create it and bails out immediately.

## Fix

Remove `app/package-lock.json` from `PKG_JSONS`. The `package.json` files are sufficient for change detection; the lockfile is an *output* of `npm install`, not a valid *input* to it.

```makefile
PKG_JSONS    := $(shell find $(APP_DIR)/packages -name 'package.json' 2>/dev/null) \
                $(APP_DIR)/package.json
```

Alternatively, commit `package-lock.json` (standard practice for reproducible installs), which also satisfies the prerequisite — but fixing the Makefile is cleaner since the circular dependency is logically wrong regardless.

--- a/Makefile
+++ b/Makefile
@@ -17,8 +17,7 @@ WEB_SRCS     := $(shell find $(APP_DIR)/packages/web/src     -name '*.ts' -o -na
                 $(APP_DIR)/packages/web/vite.config.ts
 LAMBDA_SRCS  := $(shell find $(APP_DIR)/packages/lambda      -name '*.ts' -o -name '*.mjs' 2>/dev/null)
 PKG_JSONS    := $(shell find $(APP_DIR)/packages             -name 'package.json' 2>/dev/null) \
-                $(APP_DIR)/package.json \
-                $(APP_DIR)/package-lock.json
+                $(APP_DIR)/package.json
 TS_CONFIGS   := $(APP_DIR)/tsconfig.json \
                 $(APP_DIR)/tsconfig.base.json \
                 $(shell find $(APP_DIR)/packages -name 'tsconfig*.json' 2>/dev/null)

